packages-common/pluggable-widgets-tools/tests/commands.js

```diff
@@ -0,0 +1,175 @@
+const { Mutex } = require("async-mutex");
+const { exec } = require("child_process");
+const fkill = require("fkill");
+const { copy, readJson, writeJson } = require("fs-extra");
+const { join } = require("path");
+const { mkdir, rm, tempdir } = require("shelljs");
+const { promisify } = require("util");
+const { run: runYeoman } = require("yeoman-test");
+
+const LIMIT_TESTS = process.argv.includes("--limited");
+
+const CONFIGS = [
+    ["web", "ts", "8.0"],
+    ["native", "js", "8.1"],
+    ["web", "js", "8.3"],
+    ["native", "ts", "8.6"],
+    ["web", "ts", "8.6"],
+    ["web", "js", "8.7"],
+    ["web", "ts", "latest"],
+    ["native", "js", "latest"]
+];
+
+if (LIMIT_TESTS) {
+    CONFIGS.splice(1, CONFIGS.length - 2); // Remove all configs except the first and the last
+}
+
+const mutex = new Mutex();
+
+main().catch(e => {
+    console.error(e);
+    process.exit(1);
+});
+
+async function main() {
+    console.log("Preparing...");
+
+    const { stdout: packOutput } = await execAsync("npm pack --loglevel=error", join(__dirname, ".."));
+    const toolsPackagePath = join(__dirname, "..", packOutput.trim());
+
+    const failures = (
+        await Promise.all(
+            CONFIGS.map(async config => {
+                try {
+                    await runTest(...config);
+                    return undefined;
+                } catch (e) {
+                    return [config, e];
+                }
+            })
+        )
+    ).filter(f => f);
```

Comment:
Why do you call filter here?

Reply:
Because i care only about failed cases, for example below i do `.length` check + error reporting.